### DISCUSSION TITLE
Where is Local Index saved?

### DISCUSSION THREAD

**Discussion by OnlinePage** [aurelio-labs/semantic-router#217] [Q&A]
Hi,, i have been experimenting with `semantic-router`, now as observed it uses `LocalIndex`, I am curious that whether with local index, can we store the local index at certain path? 

As observed with local index: indexes for routes are created every time we rerun the program. (This is frustrating) 
@jamescalam

**Comment by octocat20** [comment]
LocalIndex is in-memory only. It is rebuilt on each run by design, so there is no default on-disk path to reuse.

If you want persistence, use a remote index (Qdrant, Pinecone, Postgres, etc.) or save your own embeddings and reload them. For a quick local loop, rebuilding is normal; for production, switch the index backend.
